pkg/storage/util.go

```diff
@@ -56,6 +58,10 @@ func NamespaceKeyFunc(prefix string, obj runtime.Object) (string, error) {
 	if err != nil {
 		return "", err
 	}
+	name := meta.Name()
+	if ok, msg := validation.ValidatePathSegmentName(name, false); !ok {
```

Comment:
DNS1123Subdomain is too strict here. There can be valid addressable object names that are not subdomains